consensus/consensus-types/src/block_retrieval.rs

```diff
@@ -124,8 +148,20 @@ impl BlockRetrievalRequestV2 {
         self.target_round
     }
 
-    pub fn match_target_round(&self, round: u64) -> bool {
-        round <= self.target_round()
+    /// If there are gaps in the rounds, there might not be a block that stops exactly at the target
+    pub fn is_window_start_block(&self, block: &Block) -> bool {
+        block.round() == self.target_round()
+            || block.quorum_cert().certified_block().round() < self.target_round()
```

Comment:
I don't think this is sufficient condition? i.e. if the target round is missing but this is a block from much earlier round

Reply:
Ah, does this look better?

```
block.round() == self.target_round()
            || (block.round() > self.target_round && block.quorum_cert().certified_block().round() < self.target_round())
```